internal/controller/bmc_controller.go

```diff
@@ -88,19 +111,44 @@ func (r *BMCReconciler) reconcile(ctx context.Context, log logr.Logger, bmcObj *
 		log.V(1).Info("Skipped BMC reconciliation")
 		return ctrl.Result{}, nil
 	}
-
+	if r.shouldSkipReconciliation(bmcObj) {
+		log.V(1).Info("Skipped BMC reconciliation")
+		return ctrl.Result{
+			RequeueAfter: 5 * time.Second,
+		}, nil
+	}
+	if r.shouldResetBMC(bmcObj) {
+		log.V(1).Info("BMC needs reset, resetting", "BMC", bmcObj.Name)
+		if err := r.resetBMC(ctx, log, bmcObj, bmcAutoResetReason, bmcAutoResetMessage); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to reset BMC: %w", err)
+		}
+		log.V(1).Info("BMC reset initiated", "BMC", bmcObj.Name)
+		return ctrl.Result{
+			RequeueAfter: 10 * time.Second,
+		}, nil
+	}
+	if modified, err := r.handleAnnotionOperations(ctx, log, bmcObj); err != nil || modified {
+		return ctrl.Result{}, err
+	}
 	bmcClient, err := bmcutils.GetBMCClientFromBMC(ctx, r.Client, bmcObj, r.Insecure, r.BMCPollingOptions)
 	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("failed to create BMC client: %w", err)
+		return ctrl.Result{}, r.updateReadyConditionOnBMCFailure(ctx, bmcObj, err)
 	}
 	defer bmcClient.Logout()
 
+	if err := r.updateConditions(ctx, bmcObj, true, bmcReadyConditionType, corev1.ConditionTrue, bmcConnectedReason, "BMC is connected"); err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to set BMC connected condition: %w", err)
+	}
+	if err := r.updateConditions(ctx, bmcObj, false, bmcResetConditionType, corev1.ConditionFalse, "ResetComplete", "BMC reset is complete"); err != nil {
```

Comment:
Would it make sense to default the state of the reset condition to `Unknown`? Technically, someone could trigger a reset by hand and we don't necessarily notice that in the controller. Also, the `createIfNeeded` function argument could removed.